File: app/src/main/java/com/blog/demo/control/ViewStubActivity.java
package com.blog.demo.control;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.view.ViewStub;
import android.widget.TextView;

import com.blog.demo.R;

public class ViewStubActivity extends Activity implements View.OnClickListener {
    private ViewStub viewStub;
    private TextView tvInflate;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_control_view_stub);

        findViewById(R.id.btn_inflate).setOnClickListener(this);
        findViewById(R.id.btn_set_visibility).setOnClickListener(this);

        viewStub = findViewById(R.id.view_stub);
    }

    @Override
    public void onClick(View v) {
        if (v.getId() == R.id.btn_inflate) {
            if (tvInflate == null) {
                tvInflate = (TextView) viewStub.inflate();
            }
        } else if (v.getId() == R.id.btn_set_visibility) {
            if (viewStub.getVisibility() != View.VISIBLE) {
                viewStub.setVisibility(View.VISIBLE);
            }
        }
    }

}

File: app/src/main/java/com/blog/demo/control/ViewStubActivity.kt
package com.blog.demo.control

import android.app.Activity
import android.os.Bundle
import android.view.View
import android.view.ViewStub
import android.widget.Button
import android.widget.TextView
import com.blog.demo.R

class ViewStubActivity : Activity(), View.OnClickListener {
    private lateinit var viewStub: ViewStub
    private var tvInflate: TextView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_control_view_stub)

        findViewById<Button>(R.id.btn_inflate).setOnClickListener(this)
        findViewById<Button>(R.id.btn_set_visibility).setOnClickListener(this)
        viewStub = findViewById(R.id.view_stub)
    }

    override fun onClick(v: View) {
        if (v.id == R.id.btn_inflate) {
            if (tvInflate == null) {
                tvInflate = viewStub.inflate() as TextView
            }
        } else if (v.id == R.id.btn_set_visibility) {
            if (viewStub.visibility != View.VISIBLE) {
                viewStub.visibility = View.VISIBLE
            }
        }
    }

}